src/git/zcl_abapgit_git_porcelain.clas.abap

```diff
@@ -19,14 +19,22 @@ CLASS zcl_abapgit_git_porcelain DEFINITION
         new_objects   TYPE zif_abapgit_definitions=>ty_objects_tt,
       END OF ty_push_result .
 
-    CLASS-METHODS pull
+    CLASS-METHODS pull_by_branch
       IMPORTING
         !iv_url          TYPE string
         !iv_branch_name  TYPE string
       RETURNING
         VALUE(rs_result) TYPE ty_pull_result
       RAISING
         zcx_abapgit_exception .
+    CLASS-METHODS pull_by_commit
+      IMPORTING
+        !iv_url          TYPE string
+        !iv_commit_hash  TYPE zif_abapgit_definitions=>ty_sha1 OPTIONAL
```

Comment:
should this be OPTIONAL ?  I'm not sure what should happen if pulling the empty commit, it will never exist?

Reply:
Uups - thanks, missed it :(

Reply:
Hmm...I'm also curious if we would be able now to merge upload_pack_by_branch and upload_pack_by_commit into one single method - the only caller of upload_pack_by_branch uses get_branch_name for iv_branch_name

Reply:
But couldn't we use get_sha1_remote to get the branches hash instead?

Reply:
Ok, nevermind - it was me just thinking aloud ;) wouldn't be possible - we don't initially know of course the branches hash.